### PR TITLE
Confenum 9

### PR DESCRIPTION
**Pull request by fluidvanadium**
this removes ambiguous confirmation heights

### CODE REVIEW

**Review by Oscar-Pepper — CHANGES_REQUESTED** [review]
this is really great work, got some little changes / comments

File: zingo-status/src/confirmation_status.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
+use zcash_primitives::consensus::BlockHeight;
+
+/// A 32 wide bitmask with 0 in the last 5 places
+#[allow(clippy::all)]
```
[the remark this answers is not in the document]
Comment (fluidvanadium) [reply zingo-status/src/confirmation_status.rs]: fixed

File: zingolib/src/wallet/transactions.rs [hunk 2]
```diff
@@ -128,13 +129,22 @@ impl TransactionMetadataSet {
...
+                    Some(w_height) => match highest {
+                        None => Some((w.txid, w_height)),
+                        Some(highest_tuple) => {
+                            if highest_tuple.1 > w_height {
+                                highest
+                            } else {
+                                Some((w.txid, w_height))
+                            }
+                        }
+                    },
```
[the remark this answers is not in the document]
Comment (fluidvanadium) [reply zingolib/src/wallet/transactions.rs:147]: tru. alright

[the remark this answers is not in the document]
Comment (fluidvanadium) [reply zingolib/src/wallet/transactions.rs:147]: refactor was necessary to write unit test. see https://github.com/zingolabs/zingolib/pull/754

File: Cargo.toml [hunk 3]
```diff
@@ -31,6 +31,7 @@ http-body = "0.4.4"
 tonic = {version = "0.7.0", features = ["tls", "tls-roots", "tls-webpki-roots"]}
 prost = "0.10.0"
 tower = { version = "0.4" }
+json = "0.12.4"
```
Comment (AloeareV) [inline Cargo.toml:34]: Non-blocking:

`json` and `serde_json` are incompatible. Currently, we're using both, somewhat incoherently...I think serde_json is better supported, with a nicer interface...why are we adding dependencies on json?

**Review by AloeareV — DISMISSED** [review]
Approved, once comments are addressed.

File: Cargo.toml (hunk 3 above)
Comment (fluidvanadium) [reply Cargo.toml:34]: fixed

File: zingolib/src/wallet.rs [hunk 4]
```diff
@@ -1597,7 +1597,9 @@ impl LightWallet {
         #[allow(clippy::type_complexity)]
         let filters: &[Box<dyn Fn(&&D::WalletNote, &TransactionMetadata) -> bool>] =
             &[Box::new(|nnmd, transaction| {
-                transaction.block_height > BlockHeight::from_u32(anchor_height)
```
[the remark this answers is not in the document]
Comment (fluidvanadium) [reply zingolib/src/wallet.rs:1600]: aha
this is about number of confirmations too.
fixing

File: zingolib/src/wallet/transactions.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Oscar-Pepper) [reply zingolib/src/wallet/transactions.rs:147]: great! i think this is one of the powers of unit tests is it forces us to simplify the code and separate concerns of functions!

**Pull request closed without merging**